src/front/spv/function.rs

```diff
@@ -438,3 +510,79 @@ impl<I: Iterator<Item = u32>> super::Parser<I> {
         Ok(())
     }
 }
+
+fn lower(
```

Comment:
It would be hard to take by value since there are some cases that need to lower multiple expressions

Reply:
Ah wait this wouldn't work because the main loop borrows bodies so if we instead had everything in the context we wouldn't be able to call it recursively inside the loop
```rust
    fn lower(&mut self, body_idx: usize) -> crate::Block {
        let mut block = crate::Block::new();

        for item in self.bodies[body_idx].iter() { // <-- borrows self here
// -- snip --
                super::Body::Loop { body, continuing } => {
                    let body = self.lower(body); // <-- attempts to mutably borrow here
                    let continuing = self.lower(continuing);

                    block.push(
                        crate::Statement::Loop { body, continuing },
                        crate::Span::Unknown,
                    )
                }
// -- snip --
        }
    }
```